### ISSUE TITLE
Way to disable a rule?

### ISSUE THREAD

**Issue by adamjuhasz** [deadpendency/deadpendency#21]
On the usage page (https://deadpendency.com/usage) you document how to change rule settings, is there a way to disable a rule?

```yaml
rules-config:
  single-recent-author
    disable
```

For example, Haskell has many `single-recent-author` libs and more mature libs requiring less commits per year, so is it possible to disable a specific rule globally vs disabling a lib?

**Comment by adamjuhasz** [comment]
Follow up, is there a way disable / configure rules for specific libs vs globally?

**Comment by AlistairB** [comment]
Thanks again. I've been planning this, but will prioritize this now. I suggest it can be disabled as follows:

```yaml
rules-config:
  single-recent-author: false
```

I will create a separate issue for disabling / configuring rules per library. That is also planned, but will probably come a bit later.

**Comment by adamjuhasz** [comment]
We're getting errors trying to disable some rules

`Deadpendency config was invalid. Reason: expected NoRecentCommitConfig instead of !!bool`

<img width="692" alt="image" src="https://user-images.githubusercontent.com/757940/108653861-6689b680-747c-11eb-801e-79452300c36c.png">

```yaml
rules-config:
  single-recent-author: false
  no-recent-commit: false
  few-yearly-commits: false
```

**Comment by AlistairB** [comment]
@adamjuhasz sorry to clarify, it does not currently support rule disabling. I should have this work done in a day or two.

**Comment by adamjuhasz** [comment]
No rush!

**Comment by AlistairB** [comment]
@adamjuhasz I've released support for this as part of `0.3.1.0`.

I went down a bit of rabbit hole with this :sweat_smile: , but the new config features are detailed https://deadpendency.com/usage#configure-checks. The relevant change here is you can disable rules by setting them to `disabled`. So for your case the below should have the desired effect.

```yaml
rules-config:
  single-recent-author: disabled
  no-recent-commit: disabled
  few-yearly-commits: disabled
```

The other thing I have added is the ability to set rules to `warn` or `fail` (ie. you want deprecation to only produce a warning, not a failure).

BTW, I have also made the change to split the report by language, when multiple languages are detected (#22). I plan to work on linking to the dependency files in the report next week (#23).

Anyway, closing this. Feel free to reopen if you have issues or ideas on how to improve the solution. Thanks.
